deps/streamsearch/sbmh.js

```diff
@@ -24,66 +24,76 @@
  * Based heavily on the Streaming Boyer-Moore-Horspool C++ implementation
  * by Hongli Lai at: https://github.com/FooBarWidget/boyer-moore-horspool
  */
-var EventEmitter = require('events').EventEmitter,
-    inherits = require('util').inherits;
+const EventEmitter = require('events').EventEmitter,
+  inherits = require('util').inherits;
 
 function SBMH(needle) {
-  if (typeof needle === 'string')
+  if (typeof needle === 'string') {
     needle = Buffer.from(needle);
-  var i, j, needle_len = needle.length;
+  }
+
+  if ((needle instanceof Buffer) === false) {
```

Comment:
would this check pass for a string?

Reply:
In the 4 lines above a `string` gets converted into a `Buffer` using `Buffer.from`, so at this line `needle` would already be a `Buffer` if it was passes as a string:

<img width="487" alt="image" src="https://user-images.githubusercontent.com/189580/144016102-c4c435ee-4f7b-44fa-9e5f-43c41ff7fc67.png">